sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -407,13 +407,38 @@ case class GpuSortArray(base: Expression, ascendingOrder: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): cudf.ColumnVector = {
     val isDescending = isDescendingOrder(rhs)
-    lhs.getBase.listSortRows(isDescending, true)
+    val base = lhs.getBase()
+    dataType match {
+      case ArrayType(FloatType | DoubleType, _) =>
+        withResource(base.getChildColumnView(0)) {child =>
+          withResource(child.normalizeNANsAndZeros()) {normalizedChild =>
+            withResource(base.replaceListChild(normalizedChild)) {normalizedList =>
+              normalizedList.listSortRows(isDescending, true)
+            }
+          }
+        }
+      case _ =>
+        base.listSortRows(isDescending, true)
```

Comment:
I'm a bit confused. If `dataType` is not array then how can we call `base.listSortRows`? I thought that this is only callable for `ArrayType`, isn't it?

Reply:
We have the `checkInputDatatype` function to make sure the data type can only be `ArrayType`
```scala
  override def checkInputDataTypes(): TypeCheckResult = base.dataType match {
    case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
      ascendingOrder match {
        // replace Literal with GpuLiteral here
        case GpuLiteral(_: Boolean, BooleanType) =>
          TypeCheckResult.TypeCheckSuccess
        case order =>
          TypeCheckResult.TypeCheckFailure(
            s"Sort order in second argument requires a boolean literal, but found $order")
      }
    case ArrayType(dt, _) =>
      val dtSimple = dt.catalogString
      TypeCheckResult.TypeCheckFailure(
        s"$prettyName does not support sorting array of type $dtSimple which is not orderable")
    case dt =>
      TypeCheckResult.TypeCheckFailure(s"$prettyName only supports array input, but found $dt")
  }
```

Reply:
Even if that's the case, I think we should make sure the type is correct here again to reduce dependency. How about this:
```
case ArrayType(FloatType | DoubleType, _) => ....
case ArrayType(_, _) => ....
case _ => throw new [SomeTypeException]("...")

```

Reply:
Actually, we have done the data type checking in GpuOverrides.scala: https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L2771-L2779

So that we can always make sure that the value of `DataType` is `ArrayType(...)` when calling `GpuSortArray`.

The `checkInputDataTypes()` in `GpuSortArray` is to check if values are orderable, I guess. Maybe we should rename it as `checkOrderable` to avoid misunderstanding.

Reply:
I filed https://github.com/rapidsai/cudf/issues/11630 against CUDF.